mocks/datastore.go

```diff
@@ -30,25 +31,130 @@ type DataStore struct {
 	AgeDimension           bool
 }
 
+// DataStore holds the list of possible error flats along with the mocked datastore.
+// This struct can be used directly as a mock, as it implements the required methods,
+// or you can use the internal 'moq' Mock if you want ot validate calls, parameters etc.
+type DataStore struct {
+	Cfg  DataStoreConfig
+	Mock *datastoretest.DataStoreMock
+}
+
+// NewDataStore creates a new datastore mock with an empty config
+func NewDataStore() *DataStore {
+	ds := &DataStore{
+		Cfg: DataStoreConfig{},
+	}
+	mock := &datastoretest.DataStoreMock{
+		AddFilterFunc:                    ds.AddFilter,
+		AddFilterDimensionFunc:           ds.AddFilterDimension,
+		AddFilterDimensionOptionFunc:     ds.AddFilterDimensionOption,
+		AddFilterDimensionOptionsFunc:    ds.AddFilterDimensionOptions,
+		CreateFilterOutputFunc:           ds.CreateFilterOutput,
+		GetFilterFunc:                    ds.GetFilter,
+		GetFilterDimensionFunc:           ds.GetFilterDimension,
+		GetFilterOutputFunc:              ds.GetFilterOutput,
+		RemoveFilterDimensionFunc:        ds.RemoveFilterDimension,
+		RemoveFilterDimensionOptionFunc:  ds.RemoveFilterDimensionOption,
+		RemoveFilterDimensionOptionsFunc: ds.RemoveFilterDimensionOptions,
+		UpdateFilterFunc:                 ds.UpdateFilter,
+		UpdateFilterOutputFunc:           ds.UpdateFilterOutput,
+		AddEventToFilterOutputFunc:       ds.AddEventToFilterOutput,
+	}
+	ds.Mock = mock
+	return ds
+}
+
+// NotFound sets NotFound flag to true
+func (ds *DataStore) NotFound() *DataStore {
```

Comment:
I think this might not be **thread safe** - if several tests use the same mock in parallel it could end up with funky / inconsistent test results. I think the solution would be to add a lock around each field update.

**However**, that it a lot of additional work that so maybe for now just state this in the struct Godoc? If anyone runs into that issue at a later time they can fix when its required. Feels like a suitable compremise.